addons/website_slides/models/slide_channel.py

```diff
@@ -570,40 +649,81 @@ def action_channel_invite(self):
             'res_model': 'slide.channel.invite',
             'target': 'new',
             'context': local_context,
+            'name': _('Share a Course')
         }
 
     def action_add_member(self, **member_values):
-        """ Adds the logged in user in the channel members.
-        (see '_action_add_members' for more info)
-
-        Returns True if added successfully, False otherwise."""
-        return bool(self._action_add_members(self.env.user.partner_id, **member_values))
-
-    def _action_add_members(self, target_partners, **member_values):
-        """ Add the target_partner as a member of the channel (to its slide.channel.partner).
-        This will make the content (slides) of the channel available to that partner.
+        """ Enroll the logged in user in the course. (see '_action_add_members' for more info)
+
+        If the logged in user has been invited, then we sudo the add_members method
+        since we consider he has been granted access to the course.
+
+        Returns True if added and enrolled successfully, False otherwise."""
+        added_to_invited_course = False
+        courses_invited = self.filtered(lambda course: course.is_member_invitation_pending)
+        if courses_invited:
+            added_to_invited_course = bool(courses_invited.sudo()._action_add_members(self.env.user.partner_id, **member_values))
+
+        return added_to_invited_course or bool((self - courses_invited)._action_add_members(self.env.user.partner_id, **member_values))
+
+    def _action_add_members(self, target_partners, add_member_mode='enroll', **member_values):
```

Comment:
Actually I would deduce the add type based on given member_values. Something like
```
is_invite = member_values.get('member_status', 'joined') == 'invited'
```
Basically we are either in an invite (member will be 'invited' and will click on join / ), or in an addition (member will be a regular member, considered as joined and payed).

Reply:
I can, but if I introduce a boolean to make the difference between invite wizard and share wizard, it is a bit ping-pongy : use is_enroll in wizard then depending on its value, giving member_status a value, to extract it just after in _action_add_members. What do you think?